calendar.js

```diff
@@ -3,107 +3,122 @@ function myTable() {
     var table = createTable();
     getScheduleAndUpdateTable(table);
     document.body.appendChild(table);
+    
 }
 myTable();
-$("#calendar-id").on('click', 'tr', displayModal);
+$("#calendar-id").on('click', 'td', displayModal);
 
 //cancelModal is defined outside displayModal cause otherwise, it's entry is made in the event handler table each time it is 
 //encountered in the displayModal method and therefore is called multiple times on each cancel button click. 
-function cancelModal(){
+function cancelModal() {
     $("#favDialog")[0].close();
 }
 
+function getSelectedDate(todaysDate, dayOfWeekNumber) {
+    var diffInNumberOfDays = todaysDate.getDay() - dayOfWeekNumber;
+    todaysDate.setHours(-24 * diffInNumberOfDays);
+    return todaysDate;
+}
+function confirmModal(event) {
+    var selectedDate = getSelectedDate(new Date(), event.data.param1);
+    var monthNames = ["Jan", "Feb", "Mar", "Apr", "May", "Jun",
+                      "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"];
+    var epochStartTimeOfEvent = (new Date(monthNames[selectedDate.getMonth()] + " " + selectedDate.getDate() + ", " + selectedDate.getFullYear() + " " + $("#startTime option:selected").text() + ":00")).getTime()/1000.0;
+    var epochEndTimeOfEvent = (new Date(monthNames[selectedDate.getMonth()] + " " + selectedDate.getDate() + ", " + selectedDate.getFullYear() + " " + $("#endTime option:selected").text() + ":00")).getTime()/1000.0;
+    var sendInfo = {
+        eventName: $("#eventName").val(),
+        startTime: epochStartTimeOfEvent,
+        endTime: epochEndTimeOfEvent,
+        locationName: $("#locationName").val()
+    };
+
+    $.ajax({
+        url: '/',
+        method: 'POST',
+        data: JSON.stringify(sendInfo),
+        cache: false,
+        success: function (data) {
+            console.log("done!");
+        }
+    });
+    myTable();
+}
+
 function displayModal() {
- var startTime = this.innerText;
-  
-  $("#favDialog")[0].showModal(); //showModal function can be applied only to a HTMLDialogMethod and not to a jQuery object, therefore $("#favDialog")[0]
-  
- 
-  $("#startTime option").each(function() {   
-        if(startTime.includes(this.value)){
-            $("#"+this.id).attr("selected","selected");
+    var selectedStartTime = this.parentNode.innerText;
```

Comment:
please use either DOM api or Jquery consistently.. please replace `this.parentNode.innerText` with `$(this).innerText()`